packages/fork-choice/test/unit/protoArray/getCommonAncestor.test.ts

```diff
@@ -79,7 +79,11 @@ describe("getCommonAncestor", () => {
     it(`${nodeA} & ${nodeB} -> ${ancestor}`, () => {
       // biome-ignore lint/style/noNonNullAssertion: <explanation>
       const ancestorNode = fc.getCommonAncestor(fc.getNode(nodeA)!, fc.getNode(nodeB)!);
-      expect(ancestorNode && ancestorNode.blockRoot).toBe(ancestor);
+      if (ancestorNode) {
```

Comment:
this looks wrong, if anything this should be

```suggestion
      if (ancestor !== null) {
```

Reply:
That's fine. Just avoiding matching undefined to the null.

Reply:
what is if `ancestorNode` is null but it shouldn't be, we wanna match it against `ancestor`, this modifies the tests in a way were we lose confidence that `ancestorNode.blockRoot === ancestor`